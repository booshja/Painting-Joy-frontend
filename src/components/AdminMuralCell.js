// dependencies
import React from "react";
import styled from "styled-components";
// components
import { AdminTripleButton } from "../components";
import { StyledCell } from "../containers/styles/adminContainers";
import { StyledP } from "../containers/styles/adminTypography";
// breakpoints
import { breakpoints } from "../breakpoints";

const ReStyledCell = styled(StyledCell)`
    ${breakpoints("flex-direction", "", [{ 1024: "row" }])}
`;

const StyledContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
`;

const StyledMarginP = styled(StyledP)`
    margin-bottom: 0.5rem;
`;

const StyledBoldP = styled(StyledMarginP)`
    font-weight: 700;
`;

const StyledImg = styled.img`
    display: none;
    height: 100px;

    ${breakpoints("display", "", [{ 1024: "block flex" }])}
`;

const AdminMuralCell = ({
    data,
    handleArchive,
    handleUnArchive,
    handleDelete,
    showArchived,
}) => {
    const { id, title, description } = data;
    return (
        <ReStyledCell>
            <StyledContainer>
                <StyledBoldP>{title}</StyledBoldP>
                <StyledP>{description}</StyledP>
                <AdminTripleButton
                    id={id}
                    handleArchive={handleArchive}
                    handleUnArchive={handleUnArchive}
                    handleDelete={handleDelete}
                    showArchived={showArchived}
                />
            </StyledContainer>
            <StyledImg
                src={
                    process.env.REACT_APP_BACKEND_URL +
                    `murals/mural/${id}/image/1`
                }
                alt="Mural Display Image"
            />
        </ReStyledCell>
    );
};

export default AdminMuralCell;
