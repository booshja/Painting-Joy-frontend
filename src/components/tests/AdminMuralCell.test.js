import { render } from "@testing-library/react";
import AdminMuralCell from "../AdminMuralCell";

/** Smoke Test */
it("renders without crashing", () => {
    render(<AdminMuralCell />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<AdminMuralCell />);
    expect(asFragment()).toMatchSnapshot();
});
