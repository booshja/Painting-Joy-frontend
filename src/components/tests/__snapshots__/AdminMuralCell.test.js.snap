// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, showArchived false 1`] = `
<DocumentFragment>
  <div
    class="sc-gKAaRy sc-gkCoMD iYkHsM lofCdu"
  >
    <div
      class="sc-flUlpA znFni"
    >
      <p
        class="sc-iCoGMd sc-eXuyPJ sc-iGkqmO fCrOLD dbuypn ldRcAy"
      >
        This is a title
      </p>
      <p
        class="sc-iCoGMd fCrOLD"
      >
        This is a description
      </p>
      <div
        class="sc-ckTSus iukvyb"
      >
        <button
          class="sc-fbIWvP ftvQXU"
        >
          Edit
        </button>
        <button
          class="sc-FRrlG XlFNn"
        >
          Archive
        </button>
        <button
          class="sc-fXazdy llwtAi"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Mural Display Image"
      class="sc-irKDMX cJjqQT"
      src="http://localhost:5000/murals/mural/1/image/1"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showArchived true 1`] = `
<DocumentFragment>
  <div
    class="sc-gKAaRy sc-gkCoMD iYkHsM lofCdu"
  >
    <div
      class="sc-flUlpA znFni"
    >
      <p
        class="sc-iCoGMd sc-eXuyPJ sc-iGkqmO fCrOLD dbuypn ldRcAy"
      >
        This is a title
      </p>
      <p
        class="sc-iCoGMd fCrOLD"
      >
        This is a description
      </p>
      <div
        class="sc-ckTSus iukvyb"
      >
        <button
          class="sc-fbIWvP ftvQXU"
        >
          Edit
        </button>
        <button
          class="sc-FRrlG XlFNn"
        >
          Unarchive
        </button>
        <button
          class="sc-fXazdy llwtAi"
        >
          Delete
        </button>
      </div>
    </div>
    <img
      alt="Mural Display Image"
      class="sc-irKDMX cJjqQT"
      src="http://localhost:5000/murals/mural/1/image/1"
    />
  </div>
</DocumentFragment>
`;
